internal/bundleutil/bundle_test.go

```diff
@@ -0,0 +1,73 @@
+package bundleutil_test
+
+import (
+	"encoding/json"
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/operator-framework/operator-registry/alpha/declcfg"
+	"github.com/operator-framework/operator-registry/alpha/property"
+
+	"github.com/operator-framework/operator-controller/internal/bundleutil"
+)
+
+func TestGetVersion(t *testing.T) {
+	tests := []struct {
+		name        string
+		pkgProperty *property.Property
+		wantErr     error
+	}{
+		{
+			name: "valid version",
+			pkgProperty: &property.Property{
+				Type:  property.TypePackage,
+				Value: json.RawMessage(`{"version": "1.0.0"}`),
+			},
+		},
+		{
+			name: "invalid version",
+			pkgProperty: &property.Property{
+				Type:  property.TypePackage,
+				Value: json.RawMessage(`{"version": "abcd"}`),
+			},
+			wantErr: errors.New("No Major.Minor.Patch elements found"),
```

Comment:
should we test for explicit error strings / matches? I wonder if just checking that an error is returned or not is sufficient for our needs. It also makes these unit tests more resilient to changes of the error strings returned.

This is presuming we don't really care that much about the exact error string that is returned and more so just that this case results in an error.

Reply:
I agree with you on this test; the explicit message checking is probably unnecessary. For the `crdupgradesafety` tests, are you OK with the semi-explicit checking there? To me it seemed useful to do so in that context.

Reply:
@dtfranz re: crdupgradesafety tests - I'm OK with the semi explicit there if that is the only way to validate and ensure that the appropriate validation(s) are failing. In an ideal world, we would be able to have a separate test for each validation but if that isn't possible what is in this PR seems fine.